users/views.py

```diff
@@ -175,6 +175,40 @@ def profile(request):
     serializer = UserSerializer(user, many=False)
     return Response(serializer.data)
 
+@api_view(['PATCH'])
+@permission_classes((IsAuthenticated,))
+def update_skills(request): 
+    user_profile = request.user.userprofile
+    skills = request.data
+    to_set = []
+    for skill in skills:
+        try:
+            s = SkillTag.objects.get_or_create(name=skill['name'])[0]
+            to_set.append(s)
+        except Exception as e:
+            pass
+    user_profile.skills.set(to_set)
+    user_profile.save()
```

Comment:
You can add relations with `skills.add(<skill>)` one by one, so no need to create the list first. Also `skills.add` or `skills.set` already save the relations, so no need for the `user_profile.save()` call.

When using `get_or_create` it's best practice to use the unpacking and get the object and if it was created or not. In the case that you don't care if it was created (like now) the convention is to use the throwaway variable `_`.

Another good practice is to avoid generic `except` clauses for the sake of them. Also, in this case I don't think there is a possible exception. If you are trying to catch the case when there is no `name` attribute, I would recommend using a serializer to validate the payload structure and return a `ValidationError`.

Here is a suggestion:

```suggestion
    for skill in skills:
        s, _ = SkillTag.objects.get_or_create(name=skill['name'])
        user_profile.skills.add(s)
```

What do you think?

Reply:
won't the `user_profile.skills.add(s)` won't this add duplicate skills to the user?  I guess the intent with the endpoint was to send the full list of skills you want the user to be.  Is there way to do `user_profile.skills.clear` before the loop?

Reply:
Oh, yeah if you want to override all skills you can do `user_profile.skills.clear()` first to delete previous values. Or append to a list and use `.set` like you were doing. Using `.add` wont add duplicates.

You are welcome!